nodewatcher/modules/analysis/channel_allocation/allocation_algorithms.py

```diff
@@ -0,0 +1,327 @@
+            n=edge['n'],
+            w=edge['w'],
+        )
+        if edge['t'] not in known_nodes:
+            chosen_channels[edge['t']] = {
+                'freq': channel_to_frequency(edge['c']),
+                'width': edge['w'],
+            }
+
+    # Run the algorithm on an appropriate graph.
+    optimal_2ghz_graph = greedy_color_with_constraints(nx_2ghz_graph, channels_2ghz)
+    optimal_5ghz_graph = greedy_color_with_constraints(nx_5ghz_graph, channels_5ghz)
+    channel_allocations = {}
+
+    # Combine results.
+    for node in optimal_2ghz_graph:
+        channel_allocations[node] = {
+            'freq': optimal_2ghz_graph[node]['freq'],
+            'width': optimal_2ghz_graph[node]['width'],
+            'interference': optimal_2ghz_graph[node]['interference'],
+            'current_interference': optimal_2ghz_graph[node]['current_interference'],
+        }
+
+    for node in optimal_5ghz_graph:
+        channel_allocations[node] = {
+            'freq': optimal_5ghz_graph[node]['freq'],
+            'width': optimal_5ghz_graph[node]['width'],
+            'interference': optimal_5ghz_graph[node]['interference'],
+            'current_interference': optimal_5ghz_graph[node]['current_interference'],
+
+        }
+
+    return channel_allocations
+
+
+def channel_to_frequency(channel):
+    ch_to_freq = {
+        '1': 2412,
+        '2': 2417,
+        '3': 2422,
+        '4': 2427,
+        '5': 2432,
+        '6': 2437,
+        '7': 2442,
+        '8': 2447,
+        '9': 2452,
+        '10': 2457,
+        '11': 2462,
+        '36': 5180,
+        '38': 5190,
+        '40': 5200,
+        '42': 5210,
+        '44': 5220,
+        '46': 5230,
+        '48': 5240,
+        '50': 5250,
+        '52': 5260,
+        '54': 5270,
+        '56': 5280,
+        '58': 5290,
+        '60': 5300,
+        '62': 5310,
+        '64': 5320,
+        '100': 5500,
+        '102': 5510,
+        '104': 5520,
+        '106': 5530,
+        '108': 5540,
+        '110': 5550,
+        '112': 5560,
+        '114': 5570,
+        '116': 5580,
+        '118': 5590,
+        '120': 5600,
+        '122': 5610,
+        '124': 5620,
+        '126': 5630,
+        '128': 5640,
+        '132': 5660,
+        '134': 5670,
+        '136': 5680,
+        '138': 5690,
+        '140': 5700,
+        '142': 5710,
+        '144': 5720,
+        '149': 5745,
+        '151': 5755,
+        '153': 5765,
+        '155': 5775,
+        '157': 5785,
+        '159': 5795,
+        '161': 5805,
+        '165': 5825
+    }
+    return ch_to_freq[str(channel)]
+
+
+def strategy_largest_first(nx_graph):
+    """
+    Returns a list of the nodes of ``nx_graph`` in decreasing order by
+    degree.
+
+    :param nx_graph: NX Graph.
+    :return: List of nodes in nx_graph in decreasing order by degree.
+    """
+
+    return sorted(nx_graph, key=nx_graph.degree, reverse=True)
+
+
+def greedy_color_with_constraints(nx_graph, channels={}):
```

Comment:
Yep. But since we have 26 different start frequencies in the 5ghz band, this is 26^10 combinations for only 10 nodes. I was thinking of something like this:
while()
  generate a random allocation
  if it's better than the best so far:
    this is the current best
return current best.

What do you think?